cert/selfsigned_test.go

```diff
@@ -0,0 +1,65 @@
+package cert_test
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/cert"
+)
+
+// TestIsOutdatedCert checks that we'll consider the TLS certificate outdated
+// if the ip addresses or dns names don't match.
+func TestIsOutdatedCert(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "certtest")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	certPath := tempDir + "/tls.cert"
+	keyPath := tempDir + "/tls.key"
+
+	extraIPs := []string{"1.1.1.1", "123.123.123.1", "199.189.12.12"}
+	extraDomains := []string{"home", "and", "away"}
+
+	// Generate TLS files with two extra IPs and domains.
+	err = cert.GenCertPair(
+		"lnd autogenerated cert", certPath, keyPath,
+		extraIPs[:2], extraDomains[:2],
+		cert.DefaultAutogenValidity,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// We'll attempt to check up-to-date status for all variants of 1-3
+	// number of IPs and domains.
+	for numIPs := 1; numIPs <= 3; numIPs++ {
+		for numDomains := 1; numDomains <= 3; numDomains++ {
+			_, parsedCert, err := cert.LoadCert(
+				certPath, keyPath,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Using the test case's number of IPs and domains, get
+			// the outdated status of the certificate we created
+			// above.
+			outdated, err := cert.IsOutdated(
+				parsedCert, extraIPs[:numIPs],
+				extraDomains[:numDomains],
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// We expect it to be considered outdated if the IPs or
+			// domains don't match exaactly what we created.
+			expected := numIPs != 2 || numDomains != 2
+			if outdated != expected {
+				t.Fatalf("expected certificate to be "+
+					"outdated=%v", expected)
```

Comment:
nit: i'd add "got=%v" to be more informative.

Reply:
done